infra/main.bicep

```diff
@@ -330,36 +341,41 @@ module computerVision 'core/ai/cognitiveservices.bicep' = if (useGPT4V) {
 
 // Currently, we only need Key Vault for storing Computer Vision key,
 // which is only used for GPT-4V.
-module keyVault 'core/security/keyvault.bicep' = if (useGPT4V) {
+module keyVault 'core/security/keyvault.bicep' = if (useKeyVault) {
   name: 'keyvault'
   scope: keyVaultResourceGroup
   params: {
-    name: keyVaultName
+    name: !empty(keyVaultServiceName) ? keyVaultServiceName : '${abbrs.keyVaultVaults}${resourceToken}'
     location: location
     principalId: principalId
   }
 }
 
-module webKVAccess 'core/security/keyvault-access.bicep' = if (useGPT4V) {
+module webKVAccess 'core/security/keyvault-access.bicep' = if (useKeyVault) {
   name: 'web-keyvault-access'
   scope: keyVaultResourceGroup
   params: {
-    keyVaultName: keyVaultName
+    keyVaultName: keyVault.outputs.name
     principalId: backend.outputs.identityPrincipalId
   }
 }
 
-module secrets 'secrets.bicep' = if (useGPT4V) {
+module secrets 'secrets.bicep' = if (useKeyVault) {
   name: 'secrets'
   scope: keyVaultResourceGroup
   params: {
-    keyVaultName: keyVaultName
+    keyVaultName: keyVault.outputs.name
     storeComputerVisionSecret: useGPT4V
     computerVisionId: useGPT4V ? computerVision.outputs.id : ''
     computerVisionSecretName: computerVisionSecretName
+    storeSearchServiceSecret: useSearchServiceKey
+    searchServiceId: useSearchServiceKey ? searchService.outputs.id : ''
+    searchServiceSecretName: searchServiceSecretName
   }
 }
 
+
+var searchServiceSemanticRankerLevel = (searchServiceSkuName == 'free') ? 'disabled' : 'free'
```

Comment:
nit: Can we allow the customer to pass in a variable to have a standard tier semantic ranker?

Reply:
essentially just allowing AZURE_SEARCH_SEMANTIC_RANKER as a parameter in addition to being an output

Reply:
LGTM